File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        temp = dict()
        
        for key, val in enumerate(nums):
            if val in temp:
                if key - temp[val] <= k:
                    return True
                else:
                    del temp[val]
            temp[val] = key
            
        return False
        

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for(int i = 0; i < nums.length; i++) {
            Integer idx = map.get(nums[i]);
            
            if(idx != null){
                if(i-idx <= k)
                    return true;
            }
                
            map.put(nums[i], i);
        }
        
        return false;
    }
}